lib/usd/translators/shading/mtlxSymmetricShaderWriter.cpp

```diff
@@ -46,33 +45,102 @@
 
 PXR_NAMESPACE_OPEN_SCOPE
 
+/// \class MtlxUsd_SymmetricShaderWriter
+/// \brief Provides "literal" translation of Maya shading nodes to USD Shader
+/// prims that are MaterialX-compatible.
+///
+/// This shader writer performs a "literal" translation of a Maya shading node
+/// type to USD. Input and output attributes on the Maya node translate
+/// directly to inputs and outputs with the same names on the exported
+/// UsdShadeShader. With one major exception: color and alpha are kept together
+/// to solve some temporary issues about multi-output management in MaterialX/USD.
+///
+/// A static "RegisterWriter()" function is provided to simplify the
+/// registration of writers that use this class. Note however that it should be
+/// called inside a "TF_REGISTRY_FUNCTION(UsdMayaShaderWriterRegistry)" block,
+/// for example:
+///
+/// \code
+/// TF_REGISTRY_FUNCTION(UsdMayaShaderWriterRegistry)
+/// {
+///     MtlxUsd_SymmetricShaderWriter::RegisterWriter(
+///         TfToken("checker"),
+///         TfToken("MayaND_checker_color3"));
+/// };
+/// \endcode
+///
+class MtlxUsd_SymmetricShaderWriter : public MtlxUsd_BaseWriter
```

Comment:
Moved from mtlxSymmetricShaderWriter.h, which was removed. Might reappear as a separate header if we want to expose this specialization to Python, but we are in a completely different DLL, so that might require moving that code back to the mayaUsd library.